app/controllers/public_recipes_controller.rb

```diff
@@ -0,0 +1,8 @@
+class PublicRecipesController < ApplicationController
+  def index
+    @recipes = Recipe.where(public: true).order(updated_at: :desc)
+    @total_price = RecipeFood.where(recipe_id: @recipes.map(&:id)).map do |recipe_food|
+      recipe_food.food.price * recipe_food.quantity
+    end.sum
+  end
+end
```

Comment:
- [ ] We don't implement SQL Queries inside the controller, we use the model to keep the SQL queries implementation.

Reply:
Thank you. It is done. 💪